### PR TITLE
Feat/serialize import string

### LINKED ISSUE #4958: Serialization of ImportString does not have default behavior
### Initial Checks

- [X] I have searched Google & GitHub for similar requests and couldn't find anything
- [X] I have read and followed [the docs](https://pydantic-docs.helpmanual.io) and still think this feature is missing

### Description

Currently, if your model has a field whose type is `ImportString`, you will not be able to encode to a Json unless you employ a clever workaround:
```
from pydantic import BaseModel, ImportString
from types import BuiltinFunctionType


# The following class will not successfully serialize to JSON
# Since "obj" is evaluated to an object, not a pydantic `ImportString`
class WithCustomEncodersBad(BaseModel):
    obj: ImportString

    class Config:
        json_encoders = {
            ImportString: lambda x: str(x)
        }


# Create an instance
m = WithCustomEncodersBad(obj='math.cos')

try:
    m.json()
except TypeError as e:
    print(e)

# Let's do some sanity checks to verify that m.obj is not an "ImportString"
print(isinstance(m.obj, ImportString))
print(isinstance(m.obj, BuiltinFunctionType))


# So now that we know that after an ImportString is evaluated by Pydantic
# it results in its underlying object, we can configure our json encoder
# to account for those specific types
class WithCustomEncodersGood(BaseModel):
    obj: ImportString

    class Config:
        json_encoders = {
            BuiltinFunctionType: lambda x: str(x)
        }


m = WithCustomEncodersGood(obj='math.cos')
print(m.json())
```
**_We should instead implement a custom serializer to properly serialize an `ImportString`._**

> `json_encoders` like this is going in V2, see https://github.com/pydantic/pydantic-core/pull/327.
>
> Also we should probably add a custom serializer to ImportString to do this automatically.
> Depends on: https://github.com/pydantic/pydantic-core/pull/327

### Affected Components

- [ ] [Compatibility between releases](https://pydantic-docs.helpmanual.io/changelog/)
- [X] [Data validation/parsing](https://pydantic-docs.helpmanual.io/usage/models/#basic-model-usage)
- [X] [Data serialization](https://pydantic-docs.helpmanual.io/usage/exporting_models/) - `.model_dump()` and `.model_dump_json()`
- [X] [JSON Schema](https://pydantic-docs.helpmanual.io/usage/schema/)
- [ ] [Dataclasses](https://pydantic-docs.helpmanual.io/usage/dataclasses/)
- [ ] [Model Config](https://pydantic-docs.helpmanual.io/usage/model_config/)
- [ ] [Field Types](https://pydantic-docs.helpmanual.io/usage/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://pydantic-docs.helpmanual.io/usage/validation_decorator/)
- [ ] [Generic Models](https://pydantic-docs.helpmanual.io/usage/models/#generic-models)
- [ ] [Other Model behaviour](https://pydantic-docs.helpmanual.io/usage/models/) - `model_construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://pydantic-docs.helpmanual.io/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

### PR DESCRIPTION
**Pull request by ybressler**
## Change Summary
Add default serialization to `ImportString` + add tests for coverage. (Need more edge cases, but for now, is good enough.)

## Related issue number
fix #4958

## Checklist

* [X] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [ ] Documentation reflects the changes where applicable
* [ ] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [ ] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

File: pydantic/types.py [hunk 1]
```diff
@@ -290,12 +291,23 @@ def __class_getitem__(cls, item: AnyType) -> AnyType:
         def __get_pydantic_core_schema__(
             cls, source: type[Any], handler: GetCoreSchemaHandler
         ) -> core_schema.CoreSchema:
+            def serialize_import_string(v: Any) -> str:
```
Comment (samuelcolvin) [inline pydantic/types.py:294]: I guess this should be a class method.

**Review by samuelcolvin — COMMENTED** [review]
also windows tests are failing, I'll try to fix this not to include it in V2.

File: pydantic/types.py (hunk 1 above)
Comment (samuelcolvin) [reply pydantic/types.py]: done.

**Review by samuelcolvin — APPROVED** [review]

**Review by samuelcolvin — APPROVED** [review]
